src/client/Microsoft.Identity.Client/Extensibility/ManagedIdentityApplicationExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Extensibility methods for <see cref="IManagedIdentityApplication"/>
+    /// </summary>
+    public static class ManagedIdentityApplicationExtensions
+    {
+        /// <summary>
+        /// Used to determine if the currently available azure resource is able to perform Proof-of-Possession.
+        /// </summary>
+        /// <returns>Boolean indicating if Proof-of-Possession is supported</returns>
+        public static bool IsProofOfPossessionSupportedByClient(this IManagedIdentityApplication app)
```

Comment:
nit: Would it be more accurate to use the name `Is*SupportedByHost`?

Reply:
Good point, "client" is quite overloaded with meanings.